src/activity_item_handling.cpp

```diff
@@ -2846,16 +2846,24 @@ void try_fuel_fire( player_activity &act, player &p, const bool starting_fire )
             continue;
         }
 
+        // Burn just 1 item from a stack when refueling
+        item *burn_ptr = &it;
+        item it2 = burn_ptr->split( 1 );
```

Comment:
Your pointer usage is very confusing.

I suggest replacing the block (5 lines) with this:
```C++
        item it2 = it.split( 1 );
        item &burn_item = it2.is_null() ? it : it2;
```
And than use `burn_item` as the item that is getting burned.

Reply:
And another thing: this call to `item::split` removes a charge from the stack. Under *all* circumstances,.

But the removal of the last charge (which is the removal of the whole stack), only happens within the `fd.fuel_produced > last_fuel` block below (the call to `g->m.i_rem` does it).

So what happens when you put a stack of non-burning items (e.g. some nails) into the firewood stockpile? `item::split` will remove one charge from that stack, but it would never give that charge back.